k8s/k8s_test.go

```diff
@@ -0,0 +1,161 @@
+package k8s
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/pkg/api/v1"
+	hpa_apisv1 "k8s.io/client-go/pkg/apis/autoscaling/v1"
+	v1beta1 "k8s.io/client-go/pkg/apis/extensions/v1beta1"
+)
+
+var fakeK8sClient kubernetes.Interface
+
+func fakeBuilder() (kubernetes.Interface, error) {
+	return fakeK8sClient, nil
+}
+
+func TestGetCurrentNamespace(t *testing.T) {
+	expectedNS := "fakeNS"
+	fakeReadFile := func(string) ([]byte, error) {
+		return []byte(expectedNS), nil
+	}
+	ReadFile = fakeReadFile
+
+	actual, err := GetCurrentNamespace()
+	if err != nil {
+		t.Fatal("error getting current namespace", err)
+	}
+	if actual != expectedNS {
+		t.Errorf("expected %s, got %s", expectedNS, actual)
+	}
+}
+
+func TestK8sPkg(t *testing.T) {
+	var testFuncs = []struct {
+		name     string
+		testFunc func(t *testing.T)
+	}{
+		{"TestGetConfigMapData", testGetConfigMapData},
+		{"TestUpdateHPA", testUpdateHPA},
+		{"TestUpdateReplicasCount", testUpdateReplicasCount},
+		{"TestWatchConfigMap", testWatchConfigMap},
+	}
+
+	ClientBuilder = fakeBuilder
+
+	for _, tf := range testFuncs {
+		fakeK8sClient = fake.NewSimpleClientset()
```

Comment:
Can we update `fakeK8sClient` outside the loop?

Reply:
the `fake.SimpleClientSet` keeps a state, I chose to create a new one on each iteration to prevent trash and side effects of tests.

Reply:
👍